### ISSUE TITLE
documentation how to use redis cluster

### ISSUE THREAD

**Issue by glensc** [mlh758/resque-web-ish#8]
the current documentation is not useful how to configure redis cluster:
- [README.md#resque-web](https://github.com/mlh758/resque-web-ish/tree/948546301c702ecc79c0aff895f7339235de2646#resque-web)

> For all the commands below you will either need to have exported a `REDIS_CONNECTION_STRING` environment variable
> or pass that along to the `cargo run` commands when you start the app. This is the redis connection string the app
> will use when connecting.
> 
> You can also provide configuration through the following variables which will be checked if the above variable is
> not provided:
> 
> 1. REDIS_HOSTNAME: hostname for Redis database
> 2. REDIS_DATABASE: defaults to 0
> 3. REDIS_PORT: defaults to 6379
> 4. REDIS_PASSWORD

while the individual variables are self-explanatory the REDIS_CONNECTION_STRING assumes that the user must be familiar with the underlying library usage. i.e it's internal details how REDIS_CONNECTION_STRING is encoded.

Redis cluster connection is typically defined with a list of master (leader) and slave (follower) nodes, they may share ports or specified different ports individually.

for example: masters => 127.0.0.1:1234,127.0.0.1:1235

docs:
- https://redis.io/topics/cluster-tutorial

**Comment by glensc** [comment]
seems this project uses r2d2:
- https://github.com/mlh758/resque-web-ish/blob/948546301c702ecc79c0aff895f7339235de2646/src/main.rs#L35
- https://docs.rs/r2d2_redis/0.13.0/r2d2_redis/

unfortunately, there's no docs how to format the connection string:
- https://docs.rs/r2d2/0.8.8/r2d2/index.html
- https://github.com/sorccu/r2d2-redis

probably relies on another library it uses?

**Comment by glensc** [comment]
opened issue in their tracker:
- https://github.com/sorccu/r2d2-redis/issues/44

**Comment by glensc** [comment]
seems to me it can't be used out of box, some code changes needed either here or in r2d2_redis:
- https://github.com/sorccu/r2d2-redis/issues/44#issuecomment-696475092

**Comment by mlh758** [comment]
r2d2 is the wrong place to log an issue, it's just a wrapper around the redis library providing pooling of connections.

**Comment by mlh758** [comment]
The Redis crate has documentation on this https://docs.rs/redis/0.17.0/redis/cluster/index.html and it doesn't look like this library is using that feature.

**Comment by glensc** [comment]
@mlh758 see my note:
- https://github.com/mlh758/resque-web-ish/issues/8#issuecomment-696475772

referring to other note:
- https://github.com/sorccu/r2d2-redis/issues/44#issuecomment-696475092

seems to me that changes are needed in this (resque-web-ish) project if you say r2d2 is a wrapper. altho I think it needs to wrap cluster module as well.

**Comment by mlh758** [comment]
I don't think it will be too hard to add cluster support to this project. I'll want to refactor how it pulls the connection together a bit though. I think it's already a little messy, and adding another path in the config will make it more so.

I also think there is a possibility this repo _does_ actually fix the issue you've been asking about in other projects. The Resque code gets workers with [MGET](https://redis.io/commands/mget), using multiple keys at once. That's how it is encountering the CROSSLOT error. This project is sending multiple commands instead. A little less efficient, but probably not going to hit an issue requesting keys from multiple nodes in one command like that.
